lib/api/src/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -3121,6 +3153,96 @@ constructTransaction api argGenChange knownPools poolStatus apiWalletId body = d
             . Map.toList
             . foldr (uncurry (Map.insertWith (<>))) Map.empty
 
+-- When encryption is enabled we do the following:
+-- (a) find field `msg` in the object of "674" label
+-- (b) encrypt the 'msg' value if present, if there is neither "674" label
+--     nor 'msg' value inside object of it emit error
+-- (c) update value of `msg` with the encrypted initial value(s) encoded in base64
+--     [TxMetaText base64_1, TxMetaText base64_2, ..., TxMetaText base64_n]
+-- (d) add `enc` field with encryption method value 'basic'
+toMetadataEncrypted
+    :: ApiEncryptMetadata
+    -> TxMetadataWithSchema
+    -> Maybe ByteString
+    -> Either ErrConstructTx Cardano.TxMetadata
+toMetadataEncrypted apiEncrypt payload saltM = do
+    msgValue <- findMsgValue
+    msgValue' <- mapM encryptingMsg msgValue
+    pure $ updateTxMetadata msgValue'
+  where
+    pwd = BA.convert $ unPassphrase $ getApiT $ apiEncrypt ^. #passphrase
+    (secretKey, iv) = PBKDF2.generateKey metadataPBKDF2Config pwd saltM
+    getMsgValue (Cardano.TxMetaText metaField, metaValue) =
+        if metaField == "msg" then
+            Just metaValue
+        else Nothing
+    getMsgValue _ = Nothing
+    merge Nothing (Just val) = Just val
+    merge (Just val) Nothing = Just val
+    merge Nothing Nothing = Nothing
+    merge (Just _) (Just _) = error "only one 'msg' field expected"
```

Comment:
Conclusion: I think it's impossible to trigger this call to `error` from the API.

This is because the JSON serialisation for `TxMetaMap` ensures that there can never be more than one map entry with any given key.

(It's unfortunate that this restriction is not reflected in the choice of type for `TxMetaMap`, which is just a list of pairs.)